### PR TITLE
Use x/sys/unix epoll

### PR DESCRIPTION
**Pull request by dparalen**
Closes issue #736

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -196,8 +196,18 @@ else
...
-	cd vendor/github.com/osrg/gobgp/gobgp && \
-	CGO_ENABLED=0 GOARCH=$(GOARCH) GOOS=linux go build -o $(MAKEFILE_DIR)/gobgp
+ifeq "$(BUILD_IN_DOCKER)" "true"
+	@echo Building gobgp
+	$(DOCKER) run -v $(PWD)/vendor:/go/src -w /go/src/github.com/osrg/gobgp/gobgp $(DOCKER_BUILD_IMAGE) \
+    sh -c ' \
+	    GOARCH=$(GOARCH) CGO_ENABLED=0 go build -o gobgp'
+	@echo Finished building gobgp.
+else
+	pushd vendor/github.com/osrg/gobgp/gobgp && \
```
Comment (murali-reddy) [inline Makefile]: Do we need `pushd`? Why do we need push and pop here?

```sh
± make gobgp                                                                                                                                                                                                                                                                          ❯❯❯
Building for GOARCH=amd64
pushd vendor/github.com/osrg/gobgp/gobgp && \
CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -o gobgp
/bin/sh: 1: pushd: not found
Makefile:206: recipe for target 'gobgp' failed
```

with build in docker set to true I am running in to this error.

```
± make gobgp                                                                                                                                                                                                                                                                          ❯❯❯
Building for GOARCH=amd64
Building gobgp
docker run -v /home/murali/Desktop/projects/go/src/github.com/cloudnativelabs/kube-router/vendor:/go/src -w /go/src/github.com/osrg/gobgp/gobgp golang:1.10.8-alpine3.9 \
    sh -c ' \
    GOARCH=amd64 CGO_ENABLED=0 go build -o gobgp'
Finished building gobgp.
ln -f vendor/github.com/osrg/gobgp/gobgp/gobgp gobgp
ln: failed to create hard link 'gobgp' => 'vendor/github.com/osrg/gobgp/gobgp/gobgp': Operation not permitted
Makefile:200: recipe for target 'gobgp' failed
```

Could you please take a look?

Comment (dparalen) [reply Makefile]: ouch... you're right; pushd/popd aren't needed as each target rule is executed in a standalone shell instance.

Thanks for testing!

**Comment by murali-reddy** [comment]
@dparalen  I am still getting this. 

```
rm -f gobgp && ln vendor/github.com/osrg/gobgp/gobgp/gobgp gobgp
ln: failed to create hard link 'gobgp' => 'vendor/github.com/osrg/gobgp/gobgp/gobgp': Operation not permitted
```

Why cannot we just copy the binary?

**Comment by dparalen** [comment]
heh I'm complicating it I guess aren't I ;) wanted to logically make it the same thing, the binary and the Dockerfile item (which worked this way in my env)... but yeah let me just do the good old `cp -f`

@murali-reddy thanks for the testing!
 I'm updating the PR

**Comment by murali-reddy** [comment]
LGTM

**Comment by dparalen** [comment]
@murali-reddy thanks a lot!

**Pull request merged**